src/events/presenceUpdate.ts

```diff
@@ -0,0 +1,53 @@
+import { GuildMember, Presence } from 'discord.js';
+import { getDB } from '@/db';
+import { ModPingSettings } from '@/models/modPingSettings';
+
+export default async function presenceUpdateHandler(oldPresence: Presence | null, newPresence: Presence): Promise<void> {
+	if (!newPresence || !newPresence.member) return;
+	const member = newPresence.member as GuildMember;
+
+	const settings = await ModPingSettings.findOne({
+		where: { user_id: member.id }
+	});
+
+	if (!settings) return;
```

Comment:
One potential issue: if a moderator has their role removed, they will still receive the mod ping role. This should also check whether the user has a moderator role and delete the mod ping settings if not. I would suggest making a setting that's a list of moderator role IDs instead of individual settings for recruit mod/mod/head mod for simplicity.

Reply:
fair, and i'll add that

Reply:
would admin/dev also apply here?

Reply:
I think it would be fine to add admin/dev. It could be a setting called `roles.allowed-to-have-mod-ping` and be a comma-separated list of role IDs like `1234567890,0987654321,12345678901`.